### DISCUSSION TITLE
Feature Suggestions

### DISCUSSION THREAD

**Discussion by JoshuaMKW** [JoshuaMKW/Juniors-Toolbox#5] [Ideas]
This discussion acts as a fixated hub for feature requests and enhancement ideas. Feel free to share your thoughts about how this tool could be improved, and how it can be made more user friendly.

**Comment by Epicwade** [comment]
You probably allready plan on doing most of this but here's everything I could think of anyways.
- bmg templates for level titles (you have the photos for the npc dialough and since the mission titles are also bmg could be nice to have a template for that as well. Like a template for the shine select screen and one for the level fade in. Also first timers might not know where the bmg is for level titles and D.E.B.S alerts so maybe add a button to open it in the ribbon of the bmg editor or something.)
- right click context menu to see and edit an entitys parameters (id guess it would contain options like copy paste x,y,z,yaw,pitch,roll, and maybe rail)
- option for making paths or rings of coins equally spaced 
- update entity's option (button to reload entitys bmd's that way you dont need to fully reload jr's toolbox just to see weather your slight edit to a model was sucessfull)
- draw a rail (option that lets you draw a path with your mouse and a rail will be generated out of that line you drew)
- stagearc.bin editor
- automatically calculate map values for not crashing and conserving memory.
- animated textures actually animate.
- better camera data editor
- level templates (empty stage has skybox maybe a day night and sunset template that auto ajusts the object lighting values. Cant tell you how many times sunshine likes to crash when you delete stuff so if there was just a button that makes every szs empty and crash free that would be such a blessing.)
- drop to ground button so objects are always placed at the right hight.
- option to lock objects position that way you don't accidentally move it while trying to move a different thing (if possible have it save the locked property even after closing the program that way its still there when you come back a week later or something.)
- maybe an option to import and export an objects textures as png for quick and easy texture edits but that's not really nessicairy there's allready pleanty of tools for that.
- when setting g warp id's for map object chang e stage entitys have a option for the Wark id as well as the mission of said warp that way you dont have to check the ram map every time you want something to warp you to a specific mission of a world
- copying and pasting rail data from other stages.
- view collision model toggle
- streamline setting up shine and linked graphiti cameras. Also maybe for the cameras have a visual indicator of everything that's within frame that way you can know that the camera shows what you want it to look at without needing to boot up the game.
- ropes
- custom piantisimo/shadow mario path creator tool

**Reply by Bracefacemace** [reply]
No

**Comment by Bracefacemace** [comment]
Add a snap to object/bounding box feature because yes.

**Comment by Bracefacemace** [comment]
Multi object select.
